### ISSUE TITLE
Feature: Tags at the example level are not being picked up by the playwright grep

### ISSUE THREAD

**Issue by harikishanl** [vitalets/playwright-bdd#49]
**Describe the problem**
If we want to run specific tests in a feature file by the tags at the example level, the only way is to use $tags and add logic based on that. Cucumber lets us run tests at the example level by those tags but not through playwright-bdd.

**Describe the solution**
Probably similar to how we are adding the @tags to the scenario description in order to let playwright grep recognize them, it might be possible to add the tag at the example level as well

**Describe alternatives you've considered**
Alternative I have tried is to pull the cucumber tags in steps via the $tags and added logic accordingly, but this is an overhead where as in cucumber I can directly run specific tests by using the tags

**Additional context**
N/A

**Comment by vitalets** [comment]
Doesn't cli option [`--tags`](https://vitalets.github.io/playwright-bdd/#/cli?id=bddgen-test-or-just-bddgen) help with that?

For example having such a feature file:
```gherkin
Feature: Playwright Home Page

    Scenario: Check title
        Given I am on Playwright home page
        Then I see in title "Playwright"

    Scenario Outline: Check get started
        Given I am on Playwright home page
        When I click link "<link>"
        Then I see in title "<title>"

    @foo
    Examples:
        | link          | title        |
        | Get started   | Installation |    
        | API reference | Library      |  
```

I can run only examples under `@foo` tag by command:
```
npx bddgen --tags '@foo' && npx playwright test
```

**Comment by harikishanl** [comment]
Thanks @vitalets , yes I tried this earlier but it didn't work with the quotes around the tag '@foo', re-tried again today without the quotes and it worked fine, but there is a slight challenge with this approach that the bddgen only regenerates the affected feature files as spec files and the other spec files are not deleted automatically (unless I manually delete all files and run the script). So to overcome this I might need to use the test match pattern in playwright config. It would be helpful if we can rely on  just one way of tagging the tests rather than multiple ways?

**Comment by vitalets** [comment]
It should delete specs not affected by tags.
Just re-checked on [playwright-bdd-example](https://github.com/vitalets/playwright-bdd-example). if I run command without tags:
```
npx bddgen && npx playwright test
```
I get in `.features-gen`:
```
.features-gen
├── homepage.feature.spec.js
├── todopage.feature.spec.js
```
If then I run command with tags:
```
npx bddgen --tags '@todo' && npx playwright test
```
I get in `.features-gen`:
```
.features-gen
├── todopage.feature.spec.js
```
Notice `homepage.feature.spec.js` is removed.

@harikishanl 
Could you re-check on your side?

**Comment by harikishanl** [comment]
> It should delete specs not affected by tags. Just re-checked on [playwright-bdd-example](https://github.com/vitalets/playwright-bdd-example). if I run command without tags:
> 
> ```
> npx bddgen && npx playwright test
> ```
> 
> I get in `.features-gen`:
> 
> ```
> .features-gen
> ├── homepage.feature.spec.js
> ├── todopage.feature.spec.js
> ```
> 
> If then I run command with tags:
> 
> ```
> npx bddgen --tags '@todo' && npx playwright test
> ```
> 
> I get in `.features-gen`:
> 
> ```
> .features-gen
> ├── todopage.feature.spec.js
> ```
> 
> Notice `homepage.feature.spec.js` is removed.
> 
> @harikishanl Could you re-check on your side?

Hi @vitalets - Yes, I've tried it again today but the files were not deleted automatically. Also as I mentioned earlier the tags only work without the quotes around them e.g npx bddgen --tags @foo. I'm using the latest version 5.2.0.

**Comment by vitalets** [comment]
@harikishanl sound like some shell issue, could you run `npx bddgen env` and post the output?

Also I've prepared `run-example-by-tag` branch in [playwright-bdd-example](https://github.com/vitalets/playwright-bdd-example/tree/run-example-by-tag), could you to try reproduce there?

Commands:
1. `npx bddgen && npx playwright test`
2. `ls .features-gen` (should output 2 spec files)
3. `npx bddgen --tags @todo && npx playwright test` (could you also try with double quotes around @todo?)
4. `ls .features-gen` (should output 1 spec files)

Thanks in advance!

**Comment by harikishanl** [comment]
> @harikishanl sound like some shell issue, could you run `npx bddgen env` and post the output?
> 
> Also I've prepared `run-example-by-tag` branch in [playwright-bdd-example](https://github.com/vitalets/playwright-bdd-example/tree/run-example-by-tag), could you to try reproduce there?
> 
> Commands:
> 
> 1. `npx bddgen && npx playwright test`
> 2. `ls .features-gen` (should output 2 spec files)
> 3. `npx bddgen --tags @todo && npx playwright test` (could you also try with double quotes around @todo?)
> 4. `ls .features-gen` (should output 1 spec files)
> 
> Thanks in advance!

Hi @vitalets - apologies for the delay in my response. 

output from npx bddgen env:
_Playwright-bdd environment info:
platform: win32
node: v16.15.0
playwright-bdd: v5.2.0
@playwright/test: v1.36.0
@cucumber/cucumber: v9.2.0
Playwright config file: playwright.config.ts_

With the example project, in all cases with the ls command, I had 2 spec files + snapshots folder.

**Comment by vitalets** [comment]
Thank you, will investigate..

**Comment by vitalets** [comment]
Fixed in `5.3.0`. Now files should be cleared correctly on Windows.
@harikishanl could you please re-check on your side?

**Comment by harikishanl** [comment]
Hi @vitalets - the spec files not relevant for the tag are being deleted now and we can use this approach to work with tags in examples, thank you for fixing that.
